lib/Launcher.js

```diff
@@ -82,6 +82,25 @@ class Launcher {
       chromeArguments.push('--auto-open-devtools-for-tabs');
       options.headless = false;
     }
+    if (options.deterministic) {
```

Comment:
We should pass the same options to the `puppeteer.connect`: a common usecase is to launch browser in a container and connect to it remotely with `puppeteer.connect`.

It would be way better though if all these special flags could be enabled with protocol. Is this possible?

Reply:
Done - apart from a couple of caveats:

1. It doesn't work with --deterministic-fetch.  I suppose in theory  HeadlessURLRequestContextGetter::GetURLRequestContext could clear |url_request_context_| and regenerate the net::URLRequestContext but I presume that's not safe...

2. If a Pupeteer script does Puppeteer.connect _AND_ uses Browser.pages() to try and render with an existing page then this won't work.  It should work fine with new pages however.

Reply:
These are limitations that would be quite hard to debug; any chance we can assert for them?

Reply:
I've sent https://chromium-review.googlesource.com/c/chromium/src/+/939472 to add Browser.getCommandLine which lets throw an error if the expected command line flags are not set.

I've also added a flag into Puppeteer.Target to track if the target was created before or after the new Browser.deterministicMode has been set.  This is checked in Target.page() and if the browser was created in deterministic mode and we're trying to access an existing page then it will throw an error.